### PR TITLE
Bug/seasons breadcrumb fix

### PR DESCRIPTION
**Pull request by jipatino98**
### Description

Addressed the issue that was brought up by @abanuelo  [[Bug] Drill down into Division breaks app #135].(https://github.com/cascaritaco/cascarita/issues/135)

The issue arose because we were using hooks (ie `useTranslation()`) inside the crumb property inside the routes in `Routes.tsx` that get called during the render of the Breadcrumbs component. This lead to inconsistent hook ordering when the user navigated to another page.

The solution was to modify `BreadCrumb.tsx` so that the crumb was defined as a React component rather than a function that returns a React Component. Moreover, I created a new file called `BreadCrumbComponents.tsx` that will house our individual BreadCrumb component that we can call instead in our Route definitions. This not only fixed the issue but also condensed our Route file.

I also addressed the issue where if a user creates a new League, Season, etc. and they add a `/` character to the name input field, it would redirect the user to a nonexistent page. For the mean time, the user is unable to add that character to prevent the error. In the future, we may need to change how we pass the name to URL to allow users to add the `/` to their names.

### Issue Link

N/A

### Checklist

- [x] I have tested the changes locally by running `npm run test`
- [x] I have added or updated relevant documentation
- [x] I have autoformatted the code by running `npm run eslint`
- [x] I have added test cases (if applicable)

### CODE REVIEW

File: client/components/BreadCrumb/BreadCrumb.tsx [hunk 1]
```diff
@@ -11,22 +11,19 @@ interface IMatches {
...
-      const crumb = (match.handle as HandleType).crumb(
-        match.data as string | undefined,
-      );
-      return crumb as React.ReactNode;
-    });
+  const crumbs = useMemo(() => {
+    return matches
+      .filter((match) =>
+        Boolean(match.handle && (match.handle as HandleType).crumb),
+      )
```
Comment (elimanzo) [inline client/components/BreadCrumb/BreadCrumb.tsx:24]: I don't think we need the `Boolean` function if the expression can return truthy

```suggestion
      .filter(
        (match) => match.handle != null && (match.handle as HandleType).crumb,
      )
```
We may be able to remove our `IMatches` interface altogether though if we assert UIMatch annotations 

```TypeScript
  const matches = useMatches() as UIMatch<unknown, HandleType>[];

  const crumbs = useMemo(() => {
    return matches
      .filter((match) => match.handle != null && match.handle.crumb)
      .map((match) => match.handle.crumb);
  }, [matches]);
  ```

**Review by elimanzo — APPROVED** [review]
Looks good! I did have 1 nit, but not sure the right direction to take with the type assertions for `IMatches` to be more type safe

File: client/components/BreadCrumb/BreadCrumb.tsx (hunk 1 above)
Comment (jipatino98) [reply client/components/BreadCrumb/BreadCrumb.tsx]: Nice catch, I removed those the `Boolean` function and used the `UIMatch` type instead.

**Review by elimanzo — APPROVED** [review]
LGTM!

**Pull request merged**